### DISCUSSION TITLE
why  not support stm32f103

### DISCUSSION THREAD

**Discussion by ssmingyun1993** [tock/tock#3999] [Q&A] [answered]
why  not support stm32f103?

**Comment by brghena** [comment, marked as answer]
There's no general reason why we couldn't support it. We do support the STM32F303 already. https://github.com/tock/tock/tree/master/chips/stm32f303xc

One concern: some specific STM32F103 chips have very little memory. I can't remember if we have an "official" policy on required RAM, but you'd certainly need to be in the top half of the table here: https://www.st.com/en/microcontrollers-microprocessors/stm32f103.html

Finally, to test with it, some board that uses the chip would be necessary. Did you have a board in mind that you're using?

Given support for other STM32F3 and F4 chips, I suspect this would be a relatively straightforward PR for a new Tock user. Some background on Porting Tock to new chips here: https://book.tockos.org/development/porting

**Comment by alevy** [comment]
@ssmingyun1993 basically, yes please! It is both possible (and supported) to have chip and board ports out of tree, _and_ PRs for this repo (for chip and board ports) are welcome! As @brghena says, it should be reasonable to do for the cortex-m3 stm32's given the peripherals support that exists for the cortex-m4s, and there are plenty of chips in that range for which Tock could be appropriate (I'd say anything above the *8s or *Bs---so at least 64KB flash and 20KB SRAM).
